### PR TITLE
docs: added code review guide

### PR DESCRIPTION
**Pull request by bhavsarpratik**

### CODE REVIEW

File: .github/CODE_REVIEW_GUIDELINES.md
Comment (alexcg1) [inline .github/CODE_REVIEW_GUIDELINES.md:45]: do **it** like this

File: .github/CODE_REVIEW_GUIDELINES.md
Comment (alexcg1) [inline .github/CODE_REVIEW_GUIDELINES.md:67]: Do not place more than one responsibility into a single class or function, Instead, refactor into separate classes and functions.

File: .github/CODE_REVIEW_GUIDELINES.md
Comment (cristianmtr) [inline .github/CODE_REVIEW_GUIDELINES.md:23]: what's this weird character? some encoding problem?

**Review by alexcg1 — CHANGES_REQUESTED** [review]
Some tiny feedback to improve legibility

File: .github/CODE_REVIEW_GUIDELINES.md
Comment (bhavsarpratik) [reply .github/CODE_REVIEW_GUIDELINES.md:23]: Nice catch; I converted the doc to markdown through an online converter which does this for `'`

File: .github/CODE_REVIEW_GUIDELINES.md
Comment (cristianmtr) [inline .github/CODE_REVIEW_GUIDELINES.md:61]:
```suggestion
- If a `Flow()` needs to be built in the test and it's not a test in the Flow module, the test belongs under `/tests/integration`.
```

File: .github/CODE_REVIEW_GUIDELINES.md
Comment (cristianmtr) [inline .github/CODE_REVIEW_GUIDELINES.md:79]:
```suggestion
Do not create lengthy interfaces. Instead split them into smaller interfaces based on the functionality. The interface should not contain any dependencies (parameters), which are not required for the expected functionality.
```

File: .github/CODE_REVIEW_GUIDELINES.md
Comment (cristianmtr) [inline .github/CODE_REVIEW_GUIDELINES.md:83]:
```suggestion
Do not hardcode the dependencies; instead inject them.
```

File: CONTRIBUTING.md
Comment (cristianmtr) [inline CONTRIBUTING.md:74]: You can just write 

```
1. First item
1. Second item
```

and you get automatic listing.

1. First item
1. Second item

So we avoid having changes to the numbering of the items after the one you add.

**Review by cristianmtr — CHANGES_REQUESTED** [review]
Minor things

File: CONTRIBUTING.md
Comment (cristianmtr) [inline CONTRIBUTING.md:65]:
```suggestion
1. Associate your local git config with your github account. If this is your first time using git you can follow [the steps](#associate-with-github-account).
```

Comment (bhavsarpratik) [reply CONTRIBUTING.md:65]: It was originally kept 0 and so I let it be to give special importance. The rest of the numbering is working fine.

**Review by bwanglzu — APPROVED** [review]
great work, only this line `Do not merge PR without approval`. we can't do it without approval because the PR is blocked lol

File: .github/CODE_REVIEW_GUIDELINES.md
Comment (JoanFM) [inline .github/CODE_REVIEW_GUIDELINES.md:16]: Top much esa right; I think some minor style changes quickly add up

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: .github/CODE_REVIEW_GUIDELINES.md
Comment (bhavsarpratik) [reply .github/CODE_REVIEW_GUIDELINES.md:16]: Didn't get you

Comment (JoanFM) [reply .github/CODE_REVIEW_GUIDELINES.md:16]: edited

**Comment by alexcg1** [comment]
Does anyone else have feedback? If not I'll merge this

**Review by JoanFM — APPROVED** [review]

**Review by alexcg1 — APPROVED** [review]
LGTM :+1:

**Comment by alexcg1** [comment]
@hanxiao any feedback? Looks like GitHub waiting on you since you're code owner

**Review by hanxiao — APPROVED** [review]

**Pull request merged**
